packages/engine-core/src/data-proxy/utils/request.ts

```diff
@@ -89,18 +89,29 @@ async function nodeFetch(url: string, options: RequestOptions = {}): Promise<Req
   const https: typeof Https = include('https')
   const httpsOptions = buildOptions(options)
   const incomingData = [] as Buffer[]
+  const { origin } = new URL(url)
 
   return new Promise((resolve, reject) => {
     // we execute the https request and build a fetch response out of it
     const request = https.request(url, httpsOptions, (response) => {
+      // eslint-disable-next-line prettier/prettier
+      const { statusCode, headers: { location } } = response
+
+      if (statusCode! >= 301 && statusCode! <= 399 && location) {
+        if (location.startsWith('http') === false) {
+          resolve(nodeFetch(`${origin}${location}`, options))
+        } else {
+          resolve(nodeFetch(location, options))
+        }
```

Comment:
In which case `location` doesn't start with `http`?

Reply:
It can start with eg. `/page/index`, in that case, it implies the host is the original one.